### PR TITLE
Adding back the simple prefab dependency sample

### PR DESCRIPTION
**Pull request by ggfan**
this is forked from the original https://github.com/android/ndk-samples/pull/686 with minor changes ( renaming, documentation update etc )

### CODE REVIEW

File: prefab/prefab-dependecy/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+prefab-dependency
+==================
+
+C/C++ dependencies are C/C++ libraries packed with [Prefab] format into an AAR.
+This sample shows how to import [JsonCPP] AAR, one example of [C/C++ dependencies], from Google Maven to an Android project.
+
+[Prefab] could be enabled from Android Gradle Plugin 4.0+. In roder to use the feature, follow these 3 steps:
```
Comment (DanAlbert) [inline prefab/prefab-dependecy/README.md:7]: s/rorder/order/ (or just "To use...")

File: prefab/prefab-dependecy/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+* Enable the prefab feature in [gradle.properties]. **Note** that in 4.1+, this needs to be
+  done with `android.buildFeatures.prefab=true` in the application's build.gradle file.
```
Comment (DanAlbert) [inline prefab/prefab-dependecy/README.md:10]: module's, not application's (in this case it's an application, but this advice is wrong for a user working on a library module)

File: prefab/prefab-dependecy/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+CA Certificates
+----------------
+
+As explained by [this Stack Overflow
+post](https://stackoverflow.com/a/30430033/632035), the root certificates
+presented by Android since ICS are not in the format OpenSSL expects, so we need
+to provide our own certificate file. We do this by downloading curl's cacert.pem
+and storing that in our assets directory, as described in [this Stack Overflow
+post](https://stackoverflow.com/a/31521185/632035).
```
Comment (DanAlbert) [inline prefab/prefab-dependecy/README.md:69]: Not relevant.

File: prefab/prefab-dependecy/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+The C++ code in this sample can be built with either CMake (the default for this
+project) or ndk-build. To use ndk-build set the `ndkBuild` project property
+either in your `local.properties` file or on the command line by passing the
+`-PndkBuild` flag when building.
+
+To build with [Android Studio](http://developer.android.com/sdk/index.html):
+
+1. Open this project in Android Studio.
+2. Click *Run/Run 'app'*.  If you want to debug/trace code, due to [the SSL lib's known issue with lldb](https://github.com/android/ndk-samples/issues/740), make sure to apply the recommendations there for a smooth debugging experience.
```
Comment (DanAlbert) [inline prefab/prefab-dependecy/README.md:54]: Not relevant.

File: prefab/prefab-dependecy/app/build.gradle [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+
+android {
+    compileSdkVersion 29
```
Comment (DanAlbert) [inline prefab/prefab-dependecy/app/build.gradle:22]: 30?

File: prefab/prefab-dependecy/app/build.gradle [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    defaultConfig {
+        applicationId "com.example.com.example.prefabdependency"
+        minSdkVersion 21
+        targetSdkVersion 29
```
Comment (DanAlbert) [inline prefab/prefab-dependecy/app/build.gradle:26]: 30?

**Review by DanAlbert — CHANGES_REQUESTED** [review]

File: prefab/prefab-dependecy/README.md (hunk 1 above)
Comment (ggfan) [reply prefab/prefab-dependecy/README.md:7]: thx: fixed!

File: prefab/prefab-dependecy/README.md (hunk 2 above)
Comment (ggfan) [reply prefab/prefab-dependecy/README.md:10]: Ok, corrected.

File: prefab/prefab-dependecy/README.md (hunk 3 above)
Comment (ggfan) [reply prefab/prefab-dependecy/README.md:69]: removed! thanks

File: prefab/prefab-dependecy/README.md (hunk 4 above)
Comment (ggfan) [reply prefab/prefab-dependecy/README.md:54]: fixed!

File: prefab/prefab-dependecy/app/build.gradle (hunk 5 above)
Comment (ggfan) [reply prefab/prefab-dependecy/app/build.gradle:22]: yeah, updated.

File: prefab/prefab-dependecy/app/build.gradle (hunk 6 above)
Comment (ggfan) [reply prefab/prefab-dependecy/app/build.gradle:26]: updated, thanks!

**Comment by ggfan** [comment]
Fixed the issues from code review, please take one more look, thank you!

**Review by DanAlbert — APPROVED** [review]

**Pull request merged**